Storage/SQL/BookmarksSqlite.swift

```diff
@@ -247,18 +251,19 @@ public class BookmarksSqliteFactory : BookmarksModelFactory, ShareToDestination
         }, failure: failure)
     }
 
+    // Why does this function look so strangely worded? Segfault in the compiler.
     public func remove(bookmark: BookmarkNode, success: (Bool) -> (), failure: (Any) -> ()) {
         var err: NSError? = nil
-        let numDeleted = self.db.delete(&err) { (connection, err) -> Int in
-            return self.table.delete(connection, item: bookmark, err: &err)
+        let numDeleted = self.db.delete(&err) { (conn, inout err: NSError?) -> Int in
+            return self.table.delete(conn, item: bookmark, err: &err)
         }
 
-        if let err = err {
-            failure(err)
+        if err == nil {
+            success(numDeleted > 0)
```

Comment:
See the comment above! I reworded the function until the compiler stopped segfaulting.